lib/compat/wordpress-6.3/get-global-styles-and-settings.php

```diff
@@ -0,0 +1,188 @@
+			$root_selector = $block_type->supports['__experimentalSelector'];
+		} else {
+			// If no root selector found, generate default block class selector.
+			$block_name    = str_replace( '/', '-', str_replace( 'core/', '', $block_type->name ) );
+			$root_selector = ".wp-block-{$block_name}";
+		}
+
+		// Return selector if it's the root target we are looking for.
+		if ( 'root' === $target ) {
+			return $root_selector;
+		}
+
+		// If target is not `root` or `duotone` we have a feature or subfeature
+		// as the target. If the target is a string convert to an array.
+		if ( is_string( $target ) ) {
+			$target = explode( '.', $target );
+		}
+
+		// Feature Selectors ( May fallback to root selector ).
+		if ( 1 === count( $target ) ) {
+			$fallback_selector = $fallback ? $root_selector : null;
+
+			// Look for selector under `feature.root`.
+			$path = array_merge( $target, array( 'root' ) );
+
+			// Use editor specific selector if available.
+			if ( $use_editor_selectors ) {
+				$feature_selector = _wp_array_get( $block_type->editor_selectors, $path, null );
+
+				if ( $feature_selector ) {
+					return $feature_selector;
+				}
+
+				// Check if feature selector set via shorthand.
+				$feature_selector = _wp_array_get( $block_type->editor_selectors, $target, null );
+
+				// Only return if a selector was found.
+				if ( is_string( $feature_selector ) ) {
+					return $feature_selector;
+				}
+			}
+
+			// Prefer the selectors API if available.
+			if ( $has_selectors ) {
+				$feature_selector = _wp_array_get( $block_type->selectors, $path, null );
+
+				if ( $feature_selector ) {
+					return $feature_selector;
+				}
+
+				// Check if feature selector set via shorthand.
+				$feature_selector = _wp_array_get( $block_type->selectors, $target, null );
+
+				return is_string( $feature_selector ) ? $feature_selector : $fallback_selector;
+			}
+
+			// Try getting old experimental supports selector value.
+			$path             = array_merge( $target, array( '__experimentalSelector' ) );
+			$feature_selector = _wp_array_get( $block_type->supports, $path, null );
+
+			// Nothing to work with, provide fallback or null.
+			if ( null === $feature_selector ) {
+				return $fallback_selector;
+			}
+
+			// Scope the feature selector by the block's root selector.
+			$scopes    = explode( ',', $root_selector );
+			$selectors = explode( ',', $feature_selector );
+
+			$selectors_scoped = array();
+			foreach ( $scopes as $outer ) {
+				foreach ( $selectors as $inner ) {
+					$outer = trim( $outer );
+					$inner = trim( $inner );
+					if ( ! empty( $outer ) && ! empty( $inner ) ) {
+						$selectors_scoped[] = $outer . ' ' . $inner;
+					} elseif ( empty( $outer ) ) {
+						$selectors_scoped[] = $inner;
+					} elseif ( empty( $inner ) ) {
+						$selectors_scoped[] = $outer;
+					}
+				}
+			}
+
+			return implode( ', ', $selectors_scoped );
+		}
+
+		// Subfeature selector
```

Comment:
The sub-feature selector is something new, something the old experimental API didn't support, and we are introducing now. I think this is worth sharing in the developer note. See [discussion](https://github.com/WordPress/gutenberg/issues/45194#issuecomment-1293717117) for context on why it's necessary.